In [1]:
import pandas as pd 
import numpy as np 


In [2]:
df  = pd.read_html('https://en.wikipedia.org/wiki/List_of_postal_codes_of_Canada:_M')[0]
df = df[df['Borough'] != 'Not assigned'].reset_index(drop=True)
df.head(15)

Postal Code           Borough  \
0          M3A        North York   
1          M4A        North York   
2          M5A  Downtown Toronto   
3          M6A        North York   
4          M7A  Downtown Toronto   
5          M9A         Etobicoke   
6          M1B       Scarborough   
7          M3B        North York   
8          M4B         East York   
9          M5B  Downtown Toronto   
10         M6B        North York   
11         M9B         Etobicoke   
12         M1C       Scarborough   
13         M3C        North York   
14         M4C         East York   

                                        Neighbourhood  
0                                           Parkwoods  
1                                    Victoria Village  
2                           Regent Park, Harbourfront  
3                    Lawrence Manor, Lawrence Heights  
4         Queen's Park, Ontario Provincial Government  
5             Islington Avenue, Humber Valley Village  
6                                      Malvern, Rouge  
7                                           Don Mills  
8                     Parkview Hill, Woodbine Gardens  
9                            Garden District, Ryerson  
10                                          Glencairn  
11  West Deane Park, Princess Gardens, Martin Grov...  
12             Rouge Hill, Port Union, Highland Creek  
13                                          Don Mills  
14                                   Woodbine Heights

In [3]:
x = df['Neighbourhood'] == "Not assigned"
df.loc[x, 'Neighbourhood'] = df.loc[x, 'Borough']

#### Let's Find  out the shape of the dataframe, this  will help us find how many neighbourhoods we have in our dataframe. 

In [4]:
df.shape


(103, 3)

#### As we can see there is a total of 103 neighbourhoods in the dataframe. 

In [5]:
print('the shape of the dataframe is', df.shape)

the shape of the dataframe is (103, 3)


In [6]:
df1 = pd.read_csv('https://cocl.us/Geospatial_data')
df1

Postal Code   Latitude  Longitude
0           M1B  43.806686 -79.194353
1           M1C  43.784535 -79.160497
2           M1E  43.763573 -79.188711
3           M1G  43.770992 -79.216917
4           M1H  43.773136 -79.239476
..          ...        ...        ...
98          M9N  43.706876 -79.518188
99          M9P  43.696319 -79.532242
100         M9R  43.688905 -79.554724
101         M9V  43.739416 -79.588437
102         M9W  43.706748 -79.594054

[103 rows x 3 columns]

In [7]:
df = df.sort_values(by='Postal Code')
df1 = df1.sort_values(by='Postal Code')

In [8]:
latitude = df1['Latitude'].values
longitude = df1['Longitude'].values

In [9]:
df['latitude']  = latitude 
df['longitude'] = longitude 
df 

Postal Code      Borough  \
6          M1B  Scarborough   
12         M1C  Scarborough   
18         M1E  Scarborough   
22         M1G  Scarborough   
26         M1H  Scarborough   
..         ...          ...   
64         M9N         York   
70         M9P    Etobicoke   
77         M9R    Etobicoke   
89         M9V    Etobicoke   
94         M9W    Etobicoke   

                                        Neighbourhood   latitude  longitude  
6                                      Malvern, Rouge  43.806686 -79.194353  
12             Rouge Hill, Port Union, Highland Creek  43.784535 -79.160497  
18                  Guildwood, Morningside, West Hill  43.763573 -79.188711  
22                                             Woburn  43.770992 -79.216917  
26                                          Cedarbrae  43.773136 -79.239476  
..                                                ...        ...        ...  
64                                             Weston  43.706876 -79.518188  
70                                          Westmount  43.696319 -79.532242  
77  Kingsview Village, St. Phillips, Martin Grove ...  43.688905 -79.554724  
89  South Steeles, Silverstone, Humbergate, Jamest...  43.739416 -79.588437  
94                Northwest, West Humber - Clairville  43.706748 -79.594054  

[103 rows x 5 columns]

In [10]:
from geopy.geocoders import Nominatim

In [11]:
address =  'Toronto'
geolocator = Nominatim(user_agent="ny_explorer")
location = geolocator.geocode(address)
latitude = location.latitude
longitude = location.longitude

In [12]:
import folium 

In [13]:
map_toronto = folium.Map(location=[latitude, longitude], zoom_start=10)

for lat, lng, borough,  in zip(df['latitude'], df['longitude'], df['Borough']):
    label = borough
    label = folium.Popup(label, parse_html=True)
    folium.CircleMarker(
        [lat, lng],
        radius=5,
        popup=label,
        color='red',
        fill=True,
        fill_color='#3186cc',
        fill_opacity=0.7,
        parse_html=False).add_to(map_toronto)  
    
map_toronto